torchaudio/torch_overdrive.cpp

```diff
@@ -0,0 +1,52 @@
+#include <torch/extension.h>
+
+namespace torch {
+namespace audio {
+
+template <typename scalar_t>
+void overdrive_cpu_kernel(
+    at::TensorAccessor<scalar_t, 2> waveform_accessor,
+    at::TensorAccessor<scalar_t, 2> temp_accessor,
+    at::TensorAccessor<scalar_t, 1> last_in_accessor,
+    at::TensorAccessor<scalar_t, 1> last_out_accessor,
+    at::TensorAccessor<scalar_t, 2> output_waveform_accessor) {
+  int64_t n_frames = waveform_accessor.size(1);
+  int64_t n_channels = waveform_accessor.size(0);
+
+  for (int64_t i_channel = 0; i_channel < n_channels; ++i_channel) {
+    for (int64_t i_frame = 0; i_frame < n_frames; ++i_frame) {
+      last_out_accessor[i_channel] = temp_accessor[i_channel][i_frame] -
+          last_in_accessor[i_channel] + 0.995 * last_out_accessor[i_channel];
+      last_in_accessor[i_channel] = temp_accessor[i_channel][i_frame];
```

Comment:
You're setting the value of last_in to the value of temp for the current iteration so that the next iteration those values may be used . But instead you could just read from temp all the time (except for the first iteration) right? I added a similar comment for the Python code above.

Reply:
> But instead you could just read from temp all the time (except for the first iteration) right?

Yes , the first iteration needs to be handled then we can remove the last_in variable